src/web3/WalletClient.ts

```diff
@@ -564,4 +569,25 @@ export class WalletClient extends BaseClient implements IWalletClient {
     }
     return [await sender.sellRolls(txData)];
   }
+
+  /**
+   * Validate an address.
+   *
+   * @param address - The address to validate.
+   *
+   * @returns A boolean (true if the address is valid, false otherwise).
+   */
+  validateAddress(address: string): boolean {
+    address = address.slice(ADDRESS_PREFIX.length); // remove The address prefix
+    // check if the address is base58 encoded
+    try {
+      const bytesAddress = new Uint8Array(base58Decode(address));
+      if (bytesAddress.length != 32) {
```

Comment:
In the rust side they check if the length of the address, `without the version` is equal to 32

And the version is an U64 varint

@sydhds please correct me if I misunderstood something

Reply:
updated !